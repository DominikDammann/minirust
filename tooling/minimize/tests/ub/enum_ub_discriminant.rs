
extern crate intrinsics;
use intrinsics::*;

fn print_opt_bool(b: Option<bool>) {
    match b {
        None => print(-1),
        Some(false) => print(0),
        Some(true) => print(1),
    }
}

fn main() {
    unsafe {
        let x = 12u8;
        let x_ptr: *const u8 = &x;
        print_opt_bool(*(x_ptr as *const Option<bool>));
    }
}
