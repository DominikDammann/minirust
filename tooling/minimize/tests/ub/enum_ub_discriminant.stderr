fatal error: UB: load at type Enum { variants: Map({0: Variant { ty: Tuple { sized_fields: List([]), sized_head_layout: TupleHeadLayout { end: Size(1 bytes), align: Align(1 bytes) }, unsized_field: None }, tagger: Map({Size(0 bytes): (IntType { signed: Unsigned, size: Size(1 bytes) }, 2)}) }, 1: Variant { ty: Tuple { sized_fields: List([(Size(0 bytes), Bool)]), sized_head_layout: TupleHeadLayout { end: Size(1 bytes), align: Align(1 bytes) }, unsized_field: None }, tagger: Map({}) }}), discriminant_ty: IntType { signed: Signed, size: Size(8 bytes) }, discriminator: Branch { offset: Size(0 bytes), value_type: IntType { signed: Unsigned, size: Size(1 bytes) }, fallback: Known(1), children: Map({(2, 3): Known(0), (3, 256): Invalid}) }, size: Size(1 bytes), align: Align(1 bytes) } but the data in memory violates the language invariant
